### PR TITLE
fix: conditional check for oAuth client ID

### PR DESCRIPTION
**Pull request by cfex**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: apps/api/v2/src/modules/webhooks/guards/is-oauth-client-webhook-guard.ts [hunk 1]
```diff
@@ -39,14 +39,10 @@ export class IsOAuthClientWebhookGuard implements CanActivate {
       throw new BadRequestException("webhookId parameter not specified in the request");
     }
 
-    if (!webhookId) {
+    if (!oAuthClientId) {
       throw new BadRequestException("oAuthClientId parameter not specified in the request");
     }
 
-    if (!user || !webhookId || !oAuthClientId) {
```
Comment (Ryukemeister) [inline apps/api/v2/src/modules/webhooks/guards/is-oauth-client-webhook-guard.ts:46]: @cfex why did you remove this check?

**Review by Ryukemeister — APPROVED** [review]

File: apps/api/v2/src/modules/webhooks/guards/is-oauth-client-webhook-guard.ts (hunk 1 above)
Comment (keithwillcode) [inline apps/api/v2/src/modules/webhooks/guards/is-oauth-client-webhook-guard.ts:46]: Why are we getting rid of the user and webhookId checks?

Comment (Ryukemeister) [reply apps/api/v2/src/modules/webhooks/guards/is-oauth-client-webhook-guard.ts:46]: This was my my first thought as well, but if you check from line 34 to line 40 we already have checks for user and webhookId already present, and with the oAuthClientId also added I doubt if we'll ever reach this condition -`!user || !webhookId || !oAuthClientId`

Comment (cfex) [reply apps/api/v2/src/modules/webhooks/guards/is-oauth-client-webhook-guard.ts:46]: Exactly, hose lines are useless/unreachable since there is a check for each param

Comment (keithwillcode) [reply apps/api/v2/src/modules/webhooks/guards/is-oauth-client-webhook-guard.ts:46]: Ah cool hadn't seen that piece. Nice.
